components/sso-hostobject/org.wso2.carbon.hostobjects.sso/src/main/java/org/wso2/carbon/hostobjects/sso/SAMLSSORelyingPartyObject.java

```diff
@@ -402,14 +403,73 @@ public static String jsFunction_getUUID(Context cx, Scriptable thisObj,
      * @return
      * @throws Exception
      */
-    public static String jsFunction_getSAMLAuthRequest(Context cx, Scriptable thisObj,
-                                                       Object[] args,
-                                                       Function funObj)
+    public static String jsFunction_getSAMLAuthRequest(Context cx, Scriptable thisObj, Object[] args, Function funObj)
             throws Exception {
         SAMLSSORelyingPartyObject relyingPartyObject = (SAMLSSORelyingPartyObject) thisObj;
-        return Util.marshall(new AuthReqBuilder().
-                buildAuthenticationRequest(relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID)));
-
+        boolean isPassiveRequired = false;
+        String acsUrl;
+        int argLength = args.length;
+        if (argLength == 2) {
+            isPassiveRequired = (Boolean) args[0];
+            if (isPassiveRequired) {
+                acsUrl = (String) args[1];
+                if (!Boolean.valueOf(relyingPartyObject.getSSOProperty(SSOConstants.SIGN_REQUESTS))) {
+                    //builds an unsigned passive authentication request
+                    return Util.marshall(new AuthReqBuilder().
+                            buildPassiveAuthenticationRequest(relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID),
+                                    acsUrl));
+                } else {
+                    //builds a signed passive authentication request
+                    return Util.marshall(new AuthReqBuilder().
+                            buildPassiveSignedAuthenticationRequest(
+                                    relyingPartyObject.getSSOProperty(SSOConstants.ISSUER_ID),
+                                    MultitenantConstants.SUPER_TENANT_ID,
+                                    MultitenantConstants.SUPER_TENANT_DOMAIN_NAME,
+                                    relyingPartyObject.getSSOProperty(SSOConstants.IDP_URL),
+                                    acsUrl));
+                }
+            } else {
+                if (!Boolean.valueOf(relyingPartyObject.getSSOProperty(SSOConstants.SIGN_REQUESTS))) {
```

Comment:
Why have we used a Boolean.valueOf() here? shouldn't this be Boolean.parseBoolean()?